ui/chatview.go

```diff
@@ -324,14 +331,45 @@ func (chatView *ChatView) AddMessage(message *discordgo.Message) {
 	}
 }
 
+// AddDateDelimiter adds a date delimiter between messages to mark the date
+func (chatView *ChatView) AddDateDelimiter(date string, write bool) string {
+	_, _, width, _ := chatView.internalTextView.GetInnerRect()
+	dashes := (width - len(chatView.format)) / 2
+	padding := strings.Repeat("\u2500", dashes-1)
+	dateDelimiterLine := "\n[\"" + "\"]" + padding + " " + date + " " + padding
+	if write {
+		fmt.Fprint(chatView.internalTextView, dateDelimiterLine)
+	}
+	return dateDelimiterLine
+}
+
+// DateDelimiter inserts datedelimiters where needed
+func (chatView *ChatView) DateDelimiter(messages []*discordgo.Message, i int, write bool) string {
```

Comment:
* Don't use one letter variable names like `ì`, use descriptive names instead
* Remove the `write` parameter and let the caller decide whether to write the return value or not, since this adds unnecessary complexity. Or simply make a method that always writes and is being used correctly.
* `DateDelimiter` should be `GenerateDateDelimiter` or `CreateDateDelimiter`, since a method does something